### PR TITLE
Add the 3D missing surface conditions to UPwNormalFaceLoadCondition

### LINKED ISSUE #12808: [GeoMechanicsApplication] Add the 3D missing surface conditions to UPwNormalFaceLoadCondition
Add: 

template class UPwNormalFaceLoadCondition<3,6>;
template class UPwNormalFaceLoadCondition<3,8>;
~template class UPwNormalFaceLoadCondition<3,9>;~ (at least, not for now)

and ~3~ 2 test cases

### PR DESCRIPTION
**Pull request by mnabideltares**
**📝 Description**
Add missing conditions for:

template class UPwNormalFaceLoadCondition<3,6>;
template class UPwNormalFaceLoadCondition<3,8>;

**🆕 Changelog**
- Registered conditions
- Added conditions for 3D3N, 4N, 6N and 8N
- Added decumentation

### CODE REVIEW

File: applications/GeoMechanicsApplication/tests/test_conditions/Common/MaterialParametersUPw3D.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+              "DYNAMIC_VISCOSITY"        :  1e-3,
+              "THICKNESS"                :  1.0,
+              "BIOT_COEFFICIENT"         :  1.0,
+              "RETENTION_LAW"                    : "SaturatedLaw",
+              "SATURATED_SATURATION"             :  1.0,
+              "RESIDUAL_SATURATION"              :  0.06203,
+              "VAN_GENUCHTEN_AIR_ENTRY_PRESSURE" :  2.561,
+              "VAN_GENUCHTEN_GN"                 :  1.377,
+              "VAN_GENUCHTEN_GL"                 :  1.25,
+              "MINIMUM_RELATIVE_PERMEABILITY"    :  0.0001
```
Comment (rfaasse) [inline applications/GeoMechanicsApplication/tests/test_conditions/Common/MaterialParametersUPw3D.json]: If I'm correct we don't need these parameters
```suggestion
```

File: applications/GeoMechanicsApplication/tests/test_conditions/Common/MaterialParametersUPw3D.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+         },
+         "Tables": {}
```
Comment (rfaasse) [inline applications/GeoMechanicsApplication/tests/test_conditions/Common/MaterialParametersUPw3D.json]: This empty object can be removed
```suggestion
```

File: applications/GeoMechanicsApplication/tests/test_conditions/test_UPwNormalFaceLoadCondition3D3N/ProjectParameters.json [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+        }
+    },{
+        "python_module": "apply_vector_constraint_table_process",
+        "kratos_module": "KratosMultiphysics.GeoMechanicsApplication",
+        "process_name":  "ApplyVectorConstraintTableProcess",
+        "Parameters":    {
+            "model_part_name": "PorousDomain.Sides",
+            "variable_name":   "DISPLACEMENT",
+            "active":          [true,true,true],
+            "is_fixed":        [true,false,true],
```
Comment (rfaasse) [inline applications/GeoMechanicsApplication/tests/test_conditions/test_UPwNormalFaceLoadCondition3D3N/ProjectParameters.json:108]: If I read this correctly, x and z are fixed for the sides, so this means that the y direction is 'up' in the 3D schematic right?

File: applications/GeoMechanicsApplication/tests/test_conditions/documentation_data/load_flux_domain_3d.svg
Comment (rfaasse) [inline applications/GeoMechanicsApplication/tests/test_conditions/documentation_data/load_flux_domain_3d.svg:1]: If I read the ProjectParameters files and understand it correctly, the xz plane is horizontal and y is 'up'. Could you add the arrows for x/y/z so it's clear which way is which?

File: applications/GeoMechanicsApplication/tests/test_conditions/README.md [hunk 4]
```diff
@@ -30,16 +30,25 @@ The following conditions are tested.
...
+<img src="documentation_data/load_flux_domain.svg" alt="Visualization of the domain and conditions for 2D" title="Visualization of the domain and conditions for 2D" width="600">
+<img src="documentation_data/load_flux_domain_3d.svg" alt="Visualization of the domain and conditions for 3D" title="Visualization of the domain and conditions for 3D" width="600">
 
-All nodes on the sides have fixed horizontal displacements. The bottom nodes have fixed displacement in both the horizontal and the vertical
-direction. At the top of the column a vertical compressive line load with a magnitude of 1000.0 N/m (or a flux of $10^{-4} \mathrm{[m^3/s]}$ for pressure test cases) is applied. This load is kept constant during the whole analysis time. Note that gravity is absent in this computation. For cases with load conditions, the pressure is kept constant, and for cases with top fluid flux conditions, the deformation is kept constant.
+All nodes on the left and right sides have fixed horizontal displacements. The bottom nodes have fixed displacement in both the horizontal and the vertical
```
Comment (rfaasse) [inline applications/GeoMechanicsApplication/tests/test_conditions/README.md:50]: You added the part about 'left and right'. Does this mean the 'front and back' sides in the schematic are not fixed in the horizontal plane?

File: applications/GeoMechanicsApplication/tests/test_conditions/test_UPwNormalFaceLoadCondition3D3N/ProjectParameters.json [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+        "nodal_smoothing":                    false,
+        "block_builder":                      true,
+        "solution_type":                      "Quasi-Static",
+        "scheme_type":                        "Backward_Euler",
+        "reset_displacements":                true,
+        "newmark_beta":                       0.25,
+        "newmark_gamma":                      0.5,
+        "newmark_theta":                      0.5,
+        "rayleigh_m":                         0.0,
+        "rayleigh_k":                         0.0,
```
Comment (rfaasse) [inline applications/GeoMechanicsApplication/tests/test_conditions/test_UPwNormalFaceLoadCondition3D3N/ProjectParameters.json]: I think these can be cleaned up
```suggestion
```

File: applications/GeoMechanicsApplication/tests/test_conditions/test_UPwNormalFaceLoadCondition3D3N/ProjectParameters.json [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+        "strategy_type":                      "newton_raphson",
+        "convergence_criterion":              "displacement_criterion",
+        "displacement_relative_tolerance":    1.0E-4,
+        "displacement_absolute_tolerance":    1.0E-9,
+        "residual_relative_tolerance":        1.0E-4,
+        "residual_absolute_tolerance":        1.0E-9,
+        "water_pressure_relative_tolerance":  1.0E-4,
+        "water_pressure_absolute_tolerance":  1.0E-9,
```
Comment (rfaasse) [inline applications/GeoMechanicsApplication/tests/test_conditions/test_UPwNormalFaceLoadCondition3D3N/ProjectParameters.json]: Since we use displacement criterion
```suggestion
```

File: applications/GeoMechanicsApplication/tests/test_conditions/test_UPwNormalFaceLoadCondition3D3N/ProjectParameters.json [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+        "min_iterations":                     6,
+        "max_iterations":                     15,
+        "number_cycles":                      100,
```
Comment (rfaasse) [inline applications/GeoMechanicsApplication/tests/test_conditions/test_UPwNormalFaceLoadCondition3D3N/ProjectParameters.json]: I would put this to max 1 cycle
```suggestion
        "number_cycles":                      1,
```

File: applications/GeoMechanicsApplication/tests/test_conditions/test_UPwNormalFaceLoadCondition3D3N/ProjectParameters.json [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+        "reduction_factor":                   0.5,
+        "increase_factor":                    2.0,
+        "desired_iterations":                 4,
+        "max_radius_factor":                  10.0,
+        "min_radius_factor":                  0.1,
```
Comment (rfaasse) [inline applications/GeoMechanicsApplication/tests/test_conditions/test_UPwNormalFaceLoadCondition3D3N/ProjectParameters.json]: We don't use arc length, so we don't need these parameters
```suggestion
```

File: applications/GeoMechanicsApplication/tests/test_conditions/test_UPwNormalFaceLoadCondition3D3N/ProjectParameters.json [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+        "calculate_reactions":                true,
+        "max_line_search_iterations":         5,
+        "first_alpha_value":                  0.5,
+        "second_alpha_value":                 1.0,
+        "min_alpha":                          0.1,
+        "max_alpha":                          2.0,
+        "line_search_tolerance":              0.5,
```
Comment (rfaasse) [inline applications/GeoMechanicsApplication/tests/test_conditions/test_UPwNormalFaceLoadCondition3D3N/ProjectParameters.json]: We don't use line search, so these ones can be removed
```suggestion
```

File: applications/GeoMechanicsApplication/tests/test_conditions/test_UPwNormalFaceLoadCondition3D3N/ProjectParameters.json
Comment (rfaasse) [inline applications/GeoMechanicsApplication/tests/test_conditions/test_UPwNormalFaceLoadCondition3D3N/ProjectParameters.json:1]: Most comments for this parameter file also apply to the other projectparameter files

File: applications/GeoMechanicsApplication/tests/test_conditions/test_UPwNormalFaceLoadCondition3D3N/ProjectParameters.json [hunk 10]
```diff
@@ -0,0 +1,159 @@
...
+                            "GiDPostMode":           "GiD_PostAscii",
+                            "MultiFileFlag":         "SingleFile"
+                        },
+                        "file_label":          "step",
+                        "output_control_type": "step",
+                        "output_interval":     1,
+                        "body_output":         true,
+                        "node_output":         true,
+                        "skin_output":         false,
+                        "plane_output":        [],
```
Comment (rfaasse) [inline applications/GeoMechanicsApplication/tests/test_conditions/test_UPwNormalFaceLoadCondition3D3N/ProjectParameters.json]: We don't use this, so we can remove it
```suggestion
```

File: applications/GeoMechanicsApplication/tests/test_conditions/test_UPwNormalFaceLoadCondition3D3N/ProjectParameters.json [hunk 11]
```diff
@@ -0,0 +1,159 @@
...
+                        "nodal_results":       ["DISPLACEMENT","REACTION"],
+                        "gauss_point_results": []
+                    },
+                    "point_data_configuration":  []
```
Comment (rfaasse) [inline applications/GeoMechanicsApplication/tests/test_conditions/test_UPwNormalFaceLoadCondition3D3N/ProjectParameters.json]: Same here:
```suggestion
```

File: applications/GeoMechanicsApplication/tests/test_conditions.py [hunk 12]
```diff
@@ -25,11 +25,11 @@ def check_water_pressure(self, test_name, etalon_value):
         pressure = test_helper.get_water_pressure(simulation)
         self.assertAlmostEqual(etalon_value, pressure[2])
 
-    def check_displacement(self, test_name, etalon_value):
+    def check_displacement(self, test_name, etalon_value, node_index=2):
```
Comment (rfaasse) [inline applications/GeoMechanicsApplication/tests/test_conditions.py:28]: We could rename `node_index` to include that it's basically the node_index of the middle of the column, maybe something like (not sure, could be improved probably):
 ```suggestion
    def check_displacement(self, test_name, etalon_value, node_index_half_way_point=2):
```

**Review by rfaasse — COMMENTED** [review]
Thanks for adding the tests and figuring out the issues with the 8-noded condition! I have some suggestions, but none of them blocking

File: applications/GeoMechanicsApplication/tests/test_conditions/README.md (hunk 4 above)
Comment (mnabideltares) [reply applications/GeoMechanicsApplication/tests/test_conditions/README.md:50]: Right, the front and back are not fixed. I found it is unneccessary to fix, as they lead to zero displacement in z direction. It is because the poisson ratio is zet to zero

File: applications/GeoMechanicsApplication/tests/test_conditions.py (hunk 12 above)
Comment (mnabideltares) [reply applications/GeoMechanicsApplication/tests/test_conditions.py:28]: But it is not necessary to be at the half way. It can be everywhere, I just here as an optional point, selected the half way for these test cases. Don't you think to let it to be generic?

File: applications/GeoMechanicsApplication/tests/test_conditions/test_UPwNormalFaceLoadCondition3D3N/ProjectParameters.json (hunk 3 above)
Comment (mnabideltares) [reply applications/GeoMechanicsApplication/tests/test_conditions/test_UPwNormalFaceLoadCondition3D3N/ProjectParameters.json:108]: @rfaasse  Thank you for the review.
Right, x and z are fixed

File: applications/GeoMechanicsApplication/tests/test_conditions.py (hunk 12 above)
Comment (rfaasse) [reply applications/GeoMechanicsApplication/tests/test_conditions.py:28]: It seems the displacement is always checked at the half way point in this test, so we could rename it, but I'm also fine with keeping it as is.

File: applications/GeoMechanicsApplication/tests/test_conditions/README.md [hunk 13]
```diff
@@ -30,16 +30,25 @@ The following conditions are tested.
 - UPwNormalFluxCondition2D3N
 - UPwNormalFluxCondition2D4N
 - UPwNormalFluxCondition2D5N
-
+- UPwNormalFluxCondition3D3N
+- UPwNormalFluxCondition3D4N
+- UPwNormalFluxCondition3D6N
+- UPwNormalFluxCondition3D8N
```
Comment (rfaasse) [inline applications/GeoMechanicsApplication/tests/test_conditions/README.md]: The ones you registered are the normal load ones, so I guess these should be:
```suggestion
- UPwNormalFaceLoadCondition3D3N
- UPwNormalFaceLoadCondition3D4N
- UPwNormalFaceLoadCondition3D6N
- UPwNormalFaceLoadCondition3D8N
```

**Review by rfaasse — COMMENTED** [review]
Thanks for processing the review comments! There is one minor issue in the documentation of the test (you mention new flux conditions instead of normal load) that needs to be fixed, other than that, this is ready to go!

File: applications/GeoMechanicsApplication/tests/test_conditions/README.md (hunk 13 above)
Comment (mnabideltares) [reply applications/GeoMechanicsApplication/tests/test_conditions/README.md]: Done

**Review by rfaasse — APPROVED** [review]
Nice, ready to go!
